python/pyarrow/tests/test_cffi.py

```diff
@@ -627,3 +627,115 @@ def test_roundtrip_chunked_array_capsule_requested_schema():
     requested_capsule = requested_type.__arrow_c_schema__()
     with pytest.raises(NotImplementedError):
         chunked.__arrow_c_stream__(requested_capsule)
+
+
+@needs_cffi
+def test_export_import_device_array():
```

Comment:
We're copy-pasting a lot of code in those tests, can we try to reduce duplication by factoring common functionality out?